platforms/osx/build_framework.py

```diff
@@ -18,10 +18,10 @@ def getToolchain(self, arch, target):
     def getBuildCommand(self, archs, target):
         buildcmd = [
             "xcodebuild",
-            "MACOSX_DEPLOYMENT_TARGET=10.9",
+            "MACOSX_DEPLOYMENT_TARGET=10.12",
```

Comment:
Could you port patch #12921 for OSX framework in a separate PR instead ?

Reply:
Sure - will make a separate PR

Reply:
Is #12921 the proper fix for #13759 ? Looking at the patch its not immediately obvious that this is the case.

Reply:
I guess you mean to add `MACOS_DEPLOYMENT_TARGET` to the macOS build to work in the same way as `IPHONEOS_DEPLOYMENT_TARGET` in the iOS build. Correct?

Reply:
I think the real fix would be to pass a `-mmacosx-version-min` flag to the `cpu_sse41.cpp` check to prevent the build from incorrectly setting CV_SSE4_1 when the target is less than 10.12

Reply:
It would be great if you could share some link with `MACOS_DEPLOYMENT_TARGET` values difference short description (some guide how to select this value)

Yes, same way as `IPHONEOS_DEPLOYMENT_TARGET`:
- some default value
- command line parameter to change this value

BTW, Currently I'm trying to establish OSX framework on our Mac machines.

Reply:
Looks like `MACOSX_DEPLOYMENT_TARGET=10.10` [works too](http://pullrequest.opencv.org/buildbot/builders/precommit_custom_mac/builds/209) ('build' step).
10.9 doesn't (sse4.1 related messages).

Reply:
> It would be great if you could share some link with `MACOS_DEPLOYMENT_TARGET` values difference short description (some guide how to select this value)

There is a simple list of macOS versions here:
https://simple.wikipedia.org/wiki/MacOS
Most Mac users upgrade to the latest OS promptly - when choosing a macOS target it would be usual to support the latest OS version plus 2 or 3 previous versions - so right now 10.12 or 10.11 would be good values

Reply:
> Looks like `MACOSX_DEPLOYMENT_TARGET=10.10` [works too](http://pullrequest.opencv.org/buildbot/builders/precommit_custom_mac/builds/209) ('build' step).
> 10.9 doesn't (sse4.1 related messages).

For me even 10.9 is working - looks like the check for SSE4.1 support picks up the deployment target value, falis and then undefines HAVE_CPU_SSE4_1_SUPPORT